### ISSUE TITLE
Script functions called from the Lua console are executed twice and give an error

### ISSUE THREAD

**Issue by ThunderAxe31** [TASEmulators/BizHawk#1967]
1. Open any game and launch this Lua script:
```
function my_test()
	console.log("Hello")
end

while true do
	emu.frameadvance()
end
```
2. While the script is active, enter the following line in the Lua console and hit enter: `my_test()`

3. The following output will appear in the Lua console: 
```
Hello
NLua.Exceptions.LuaException: unprotected error in call to Lua API (0)
   at NLua.Lua.PanicCallback(IntPtr luaState)
   at lua_error(lua_State* )
   at NLua.ObjectTranslator.throwError(IntPtr luaState, Object e)
   at NLua.Lua.SetPendingException(Exception e)
   at NLua.LuaMethodWrapper.call(IntPtr luaState)
   at NLua.MetaFunctions.runFunctionDelegate(IntPtr luaState)
   at lua_pcall(lua_State* , Int32 , Int32 , Int32 )
   at NLua.Lua.DoString(String chunk, String chunkName)
   at BizHawk.Client.EmuHawk.Win32LuaLibraries.ExecuteString(String command)
   at BizHawk.Client.Common.LuaSandbox.Sandbox(Action callback, Action exceptionCallback)Hello
```
It's weird. The function is called twice, and in between you get an error... however script execution doesn't get stopped, and continues without any apparent issue.

**Comment by adelikat** [comment]
To clarify, The behavior of LuaInterface vs NLua is affected here but both throw some kind of error.

However, I'm not sure this is supposed to work... I certainly wasn't an intended feature, and I don't know that it ever worked

**Comment by ThunderAxe31** [comment]
Oh right, I forgot to mention. The preinstalled functions can be used without issues. I find it a handy feature, as I always use it for retrieving the exact cycle count by inputting emu.totalexecutedcycles(). Also, if it wasn't supposed to work, then for what else was the input box was put there in first place? Anyway, if I wanted to execute one of my functions at arbitrary times I could still use input.get(), it's just that it would be less intuitive and require emulation to be unpaused.

**Comment by YoshiRulz** [comment]
CNR with NLua+KopiLua under Mono at fdbb34dff. (This only affects Windows.)

On Windows, repro'd with both engines at fdbb34dff, but CNR with new KeraLua engine at 6174398fb. So this was fixed by #3361.
